### PR TITLE
feat: add more version information

### PR DESCRIPTION
**Pull request by amphro**
### What does this PR do?
Adding More Information for Version Command.
Added new options[ --verbose , --json] to the Version command which provides information of CLI Version, Plugin Version, OS and its version
### What issues does this PR fix or reference?
@W-8682815@

### CODE REVIEW

File: src/cli.ts [hunk 1]
```diff
@@ -117,10 +118,58 @@ function debugCliInfo(version: string, channel: string, env: Env, config: IConfi
...
+    this.cliVersion = cliVersion;
+    this.pluginVersions = pluginVersions;
+    this.osVersion = osVersion;
+  }
+}
+
+class SfdxMain extends Main {
+  // Function which returns Version object which includes cli version, plugin versions, OS and its version.
+  protected getIssueVersionObject(): VersionObject {
+    const cliVersion = this.config.userAgent;
```
Comment (mshanemc) [inline src/cli.ts:136]: I'd like to see I'd want to see what's currently "cliVersion" split out into the 3 parts (cliVersion, architecture, and nodeVersion) as separate json properties

File: src/cli.ts [hunk 2]
```diff
@@ -117,10 +118,58 @@ function debugCliInfo(version: string, channel: string, env: Env, config: IConfi
...
+  }
+
+  protected _version(): never {
+    const options: Set<string> = new Set(this.argv);
+
+    // Checking if options has json or verbose
+    if (options.has('--verbose') || options.has('--json')) {
+      const versionObject: VersionObject = this.getIssueVersionObject();
+      if (options.has('--json')) {
+        this.log(`${JSON.stringify(versionObject, null, '\t')}`);
```
Comment (mshanemc) [inline src/cli.ts:153]: normally we `return` the object.  Does that work here?

File: src/cli.ts [hunk 3]
```diff
@@ -117,10 +118,58 @@ function debugCliInfo(version: string, channel: string, env: Env, config: IConfi
...
+      } else {
+        this.log(` CLI Version : \n\t${versionObject.cliVersion}`);
+        this.log('\n Plugin Version: ');
+        versionObject.pluginVersions.forEach((plugin) => {
+          this.log(`\t${plugin}`);
+        });
+        this.log(`\n OS and Version: \n\t${versionObject.osVersion}`);
+      }
+    } else {
```
Comment (mshanemc) [inline src/cli.ts:162]: just a style preference, but I'd put this first.  ex:

```
if (!options.has('--verbose') && !options.has('--json')) {
  this.log(this.config.userAgent);
  return; // assuming we don't have to use exit codes in oclif context 
}
```

That gets you out early and avoids having nested if/else (or any else at all!)

File: src/cli.ts [hunk 4]
```diff
@@ -117,10 +118,58 @@ function debugCliInfo(version: string, channel: string, env: Env, config: IConfi
   );
 }
 
+class VersionObject {
```
Comment (mshanemc) [inline src/cli.ts:121]: Why would this be a class?  Would it be cleaner as an interface that `getIssueVersionObject` could return?

File: src/cli.ts [hunk 5]
```diff
@@ -117,10 +118,58 @@ function debugCliInfo(version: string, channel: string, env: Env, config: IConfi
...
+    const pluginVersion: string = exec('sfdx plugins --core', {
+      silent: true,
+    }).toString();
+    const pluginVersions: string[] = pluginVersion.split('\n');
+    pluginVersions.pop();
+    const osVersion = `${os.type()} ${os.release()}`;
+    return new VersionObject(cliVersion, pluginVersions, osVersion);
+  }
+
+  protected _version(): never {
```
Comment (mshanemc) [inline src/cli.ts:146]: How does this get called?  Some kind of cool oclif magic?

Comment (Vamsi-Mundra-Salesforce) [reply src/cli.ts:146]: @amphro have to answer this.

File: src/cli.ts (hunk 4 above)
Comment (Vamsi-Mundra-Salesforce) [reply src/cli.ts]: Changed to Interface

File: src/cli.ts (hunk 3 above)
Comment (Vamsi-Mundra-Salesforce) [reply src/cli.ts]: Updated based on the suggestion.

File: src/cli.ts (hunk 2 above)
Comment (Vamsi-Mundra-Salesforce) [reply src/cli.ts]: Nope, it is not working here.

File: src/cli.ts (hunk 1 above)
Comment (Vamsi-Mundra-Salesforce) [reply src/cli.ts]: Updated based on suggestion.

**Review by mshanemc — APPROVED** [review]

File: src/cli.ts [hunk 6]
```diff
@@ -117,10 +118,63 @@ function debugCliInfo(version: string, channel: string, env: Env, config: IConfi
   );
 }
 
+interface VersionDetailI {
```
Comment (WillieRuemmele) [inline src/cli.ts:121]: should this be just `VersionDetail` not `VersionDetailI` with the capital I at the end?

**Review by WillieRuemmele — APPROVED** [review]

**Pull request merged**